tensorflow_transform/mappers.py

```diff
@@ -117,7 +117,7 @@ def scale_to_z_score(x, elementwise=False, name=None):
     # x_mean will be float32 or float64, depending on type of x.
     x_mean, x_var = analyzers._mean_and_var(  # pylint: disable=protected-access
         x, reduce_instance_dims=not elementwise)
-    return (tf.cast(x, x_mean.dtype) - x_mean) / tf.sqrt(x_var)
+    return tf.cond(x_var>0, lambda: (tf.cast(x, x_mean.dtype) - x_mean) / tf.sqrt(x_var), lambda: (tf.cast(x, x_mean.dtype) - x_mean))
```

Comment:
also most likely the PR has a bug....x can be tensor of shape [100] vs x can be a 200 tensor of shape[]....in both case the code should work...

Reply:
Yes, please make sure this is covered by a unittest.

Additionally, 

- let's use tf.where to account for batches, something similar to:
```
    denominator = tf.cast(x, x_mean.dtype) - x_mean
    return tf.where(denominator > 0, denominator / tf.sqrt(x_var), denominator)
```
- make sure that the line length doesn't exceed 80 characters
- we need spaces around operators, so "x_var > 0" instead of "x_var>0"

These last 2 are needed to please our internal style guide.

Reply:
In the initial revision I believe the answer is yes, but I think that tf.where will solve it (see my fixed suggestion below, I think my previous one was off).

@Debasish-Das-CK, note that the 2 cases here are:
vector - [batch_size, 200]
scalar - [batch_size]

so it's never a [] shape, and it can be a [200] shape if 200 is the batch size.

```
    numerator = tf.cast(x, x_mean.dtype) - x_mean
    denominator = tf.sqrt(x_var)
    return tf.where(denominator != 0, numerator / denominator, x_mean)
```